gammapy/scripts/make_bg_cube_models.py

```diff
@@ -0,0 +1,329 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+import os
+import logging
+import numpy as np
+from astropy.coordinates import Angle, SkyCoord
+
+from ..utils.scripts import get_parser
+from ..obs import ObservationTable, DataStore
+from .. import datasets
+from ..data import EventListDataset
+from ..background import make_bg_cube_model
+# TODO: revise imports!!!
+
+__all__ = ['make_bg_cube_models',
+           'create_bg_observation_list',
+           'group_observations',
+           'stack_observations',
+           ]
+
+
+DEBUG = 1 # 0: no output, 1: output, 2: run fast, 3: more verbose
+# TODO: remove the DEBUG global variable, when the logger works!!!
+
+def main(args=None):
+    parser = get_parser(make_bg_cube_models)
+    parser.add_argument("-l", "--log", dest="loglevel",
+                        choices=['debug', 'info', 'warning', 'error',
+                                 'critical'],
+                        help="Set the logging level")
+    parser.add_argument('fitspath', type=str,
+                        help='Dir path to input event list fits files.')
+    args = parser.parse_args(args)
+    make_bg_cube_models(**vars(args))
+
+
+def make_bg_cube_models(loglevel,
+                        fitspath):
+    """Create background cube models from the complete dataset of an experiment.
+
+    Starting with gamma-ray event lists and effective area IRFs,
+    make background templates. Steps
+
+    1. make a global event list from a datastore
+    2. filter the runs keeping only the ones far from known sources
+    3. group the runs according to similar observation conditions (i.e. alt, az)
+    4. create a bg cube model for each group using:
+        * the `~gammapy.background.make_bg_cube_model` method
+        * and `~gammapy.background.CubeBackgroundModel` objects as containers
+
+    The models are stored into FITS files.
+
+    It can take a few minutes to run.
+
+    TODO: revise doc!!!
+
+    Parameters
+    ----------
+    loglevel : str
+        Level for the logger.
+    fitspath : str
+        path to dir containing event list fits files and a list of them
+    """
+    if (loglevel):
+        logging.basicConfig(level=getattr(logging, loglevel.upper()), format='%(levelname)s - %(message)s')
+
+    create_bg_observation_list(fitspath)
+    group_observations()
+    stack_observations(fitspath)
+
+
+# define observation binning
+# TODO: store it in a file (fits? ascii?) (use an astropy table as help?)!!!
+
+# define a binning in altitude angle
+# TODO: ObservationGroups
+# https://github.com/mapazarr/gammapy/blob/bg-api/dev/background-api.py#L55
+altitude_edges = Angle([0, 20, 23, 27, 30, 33, 37, 40, 44, 49, 53, 58, 64, 72, 90], 'degree')
+if DEBUG > 1:
+    altitude_edges = Angle([0, 45, 90], 'degree')
```

Comment:
I think we and others will want to try many different ways of grouping the off runs, and binning / smoothing / fitting the cubes. So having one hard-coded config and one alternative "debug" config for quicker tests doesn't scale to this task.

Making this nice and easy to configure is a hard problem and we have to think / discuss how to do this best.
My first thought would be do have some "...Config" or "...Specification" class (or plain Python dict for now)
and then for the command line tool accept a YAML config file describing this configuration as input.

I think that's what I want to do in Gammapy for all the configurable analyses ... YAML config files.
There's many packages that do this ... here's just one example: http://fermipy.readthedocs.org/en/latest/quickstart.html#creating-a-configuration-file